Paraswap-Agents/Role-Changes/src/agent.spec.ts

```diff
@@ -0,0 +1,144 @@
+  // init the agent
+  let handler: HandleTransaction;
+  handler = provideHandleTransaction(CONTRACT);
+
+  it("should ignore transactions without events", async () => {
+    const tx: TransactionEvent = new TestTransactionEvent();
+
+    const findings: Finding[] = await handler(tx);
+
+    expect(findings).toStrictEqual([]);
+  });
+
+  it("should ignore events emitted on a different contract", async () => {
+    const differentContract = createAddress("0xd4");
+    // events generation
+    const log1 = EVENTS_IFACE.encodeEventLog(EVENTS_IFACE.getEvent("RoleAdminChanged"), [
+      formatBytes32String("abc"), // role
+      formatBytes32String("acd"), // previousAdminRole
+      formatBytes32String("ade"), // newAdminRole
+    ]);
+    const log2 = EVENTS_IFACE.encodeEventLog(EVENTS_IFACE.getEvent("RoleGranted"), [
+      formatBytes32String("bbc"), // role
+      createAddress("0xa1"), // account
+      createAddress("0xa2"), // sender
+    ]);
+    const log3 = EVENTS_IFACE.encodeEventLog(EVENTS_IFACE.getEvent("RoleRevoked"), [
+      formatBytes32String("cbc"), // role
+      createAddress("0xb1"), // account
+      createAddress("0xb2"), // sender
+    ]);
+    // create a transaction with the previous event logs
+    const tx: TransactionEvent = new TestTransactionEvent()
+      .addAnonymousEventLog(differentContract, log1.data, ...log1.topics)
+      .addAnonymousEventLog(differentContract, log2.data, ...log2.topics)
+      .addAnonymousEventLog(differentContract, log3.data, ...log3.topics);
+
+    const findings: Finding[] = await handler(tx);
+
+    expect(findings).toStrictEqual([]);
+  });
+
+  it("should ignore different events on the same contract", async () => {
+    // events generation
+    const log = IRRELEVANT_EVENT_IFACE.encodeEventLog(IRRELEVANT_EVENT_IFACE.getEvent("IrrelevantEvent"), [
+      formatBytes32String("bbc"), // role
+      createAddress("0xa1"), // account
+      createAddress("0xa2"), // sender
+    ]);
+    // create a transaction with the previous event logs
+    const tx: TransactionEvent = new TestTransactionEvent().addAnonymousEventLog(CONTRACT, log.data, ...log.topics);
+    const findings: Finding[] = await handler(tx);
+
+    expect(findings).toStrictEqual([]);
+  });
+
+  it("should return single finding", async () => {
+    // events generation
+    const log1 = EVENTS_IFACE.encodeEventLog(EVENTS_IFACE.getEvent("RoleAdminChanged"), [
+      formatBytes32String("abc"), // role
+      formatBytes32String("acd"), // previousAdminRole
+      formatBytes32String("ade"), // newAdminRole
+    ]);
+    // create a transaction with the previous event logs
+    const tx: TransactionEvent = new TestTransactionEvent().addAnonymousEventLog(CONTRACT, log1.data, ...log1.topics);
+    const findings: Finding[] = await handler(tx);
+
+    expect(findings).toStrictEqual([createFinding("RoleAdminChanged", log1.topics)]);
+  });
+
+  it("should return multiple findings", async () => {
```

Comment:
Similar to the test before, I think this test could be strengthened by adding an event log with `IrrelevantEvent` as one of the middle logs (e.g. `log2` or `log3`) and confirming the agent "skip" the incorrect one and returns the three correct ones.

Also recommend not re-using test values for the event arguments.